akka-stream/src/main/scala/DocumentStage.scala

```diff
@@ -43,12 +43,10 @@ private[akkastream] class DocumentStage[T](
       @inline private def tailable = cursor.wrappee.tailable
 
       private var request: () => Future[Option[Response]] = { () =>
+        // Call onFirst immediately to make sure only one request is sent.
         cursor.makeRequest(maxDocs).andThen {
-          case Success(r) if (
-            tailable && r.reply.numberReturned > 0
-          ) => onFirst()
-
-          case Success(_) if (!tailable) => onFirst()
+          case Success(_) =>
```

Comment:
What would be the hypothesis to explain a memory issue whereas there is no data? We could think of it when there is too many data or that data are keep too long...